src/sip/strans.c

```diff
@@ -219,6 +223,7 @@ static bool cancel_handler(struct sip *sip, const struct sip_msg *msg)
 
 	case TRYING:
 	case PROCEEDING:
+		st->cancel_msg = mem_ref((void *)msg);
```

Comment:
please do mem_deref first to avoid memory leaks

Reply:
I don't quite get what this should achieve? I store a ref to this message, calling mem_deref first would effectively keep the ref counter the same right?

I tried to copy the code for `strans.msg` which does a `mem_ref` before storing it in the strans.

Reply:
if `cancel_msg` points to another object before calling, there will be a leak.

please look at this example:

```
src/http/client.c-1000- mem_deref(cli->tls);
src/http/client.c:1001: cli->tls = mem_ref(tls);
```

Reply:
Ah I see. Sorry for the confusion. I didn't assume that this codepath could be taken more than once, but better safe than sorry :+1: